pkg/buganalysis/cache.go

```diff
@@ -42,21 +45,62 @@ func NewBugCache() BugCache {
 
 // updates a global variable with the bug mapping based on current failures.
 func (c *bugCache) UpdateForFailedTests(failedTestNames ...string) error {
+	newBugs, lastUpdateError := findBugsForFailedTests(failedTestNames...)
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	for testName, bug := range newBugs {
+		if _, found := c.cache[testName]; found {
+			continue
+		}
+		c.cache[testName] = bug
+	}
+	c.lastUpdateError = lastUpdateError
+	return lastUpdateError
+}
+
+func getJobKey(jobName string) string {
+	return fmt.Sprintf("job=%v=all", jobName)
+}
+
+// updates a global variable with the bug mapping based on current failures.
+func (c *bugCache) UpdateJobBlockers(jobNames ...string) error {
+	failedTestNames := []string{}
+	for _, jobName := range jobNames {
+		failedTestNames = append(failedTestNames, getJobKey(jobName))
+	}
+	newBugs, lastUpdateError := findBugsForFailedTests(failedTestNames...)
```

Comment:
so the ci-search api does return you the BZ metadata in such a way that you could restrict this to only match bugs that have this value in the "environment" field instead of in a comment, if you wanted to reduce the risk of false positives.

probably not worth it but food for thought.